mongo/options/encryptoptions.go

```diff
@@ -34,13 +44,29 @@ func (e *EncryptOptions) SetKeyAltName(keyAltName string) *EncryptOptions {
 	return e
 }
 
-// SetAlgorithm specifies an algorithm to use for encryption. This should be AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic
-// or AEAD_AES_256_CBC_HMAC_SHA_512-Random. This is required.
+// SetAlgorithm specifies an algorithm to use for encryption. This should be one of the following:
+// - AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic
+// - AEAD_AES_256_CBC_HMAC_SHA_512-Random
+// - Indexed
+// - Unindexed
+// This is required.
 func (e *EncryptOptions) SetAlgorithm(algorithm string) *EncryptOptions {
 	e.Algorithm = algorithm
 	return e
 }
 
+// SetQueryType specifies the intended query type. It is only valid to set if algorithm is "Indexed".
+func (e *EncryptOptions) SetQueryType(queryType *QueryType) *EncryptOptions {
+	e.QueryType = queryType
+	return e
+}
+
+// SetContentionFactor specifies the contention factor. It is only valid to set if algorithm is "Indexed".
+func (e *EncryptOptions) SetContentionFactor(contentionFactor *int64) *EncryptOptions {
+	e.ContentionFactor = contentionFactor
```

Comment:
Similar to above, is there a reason to accept `*int64` instead of `int64`? Can we take the pointer value in `SetContentionFactor` instead?

E.g.
```go
func (e *EncryptOptions) SetContentionFactor(contentionFactor int64) *EncryptOptions {
	e.ContentionFactor = &contentionFactor
	// ...
```

Reply:
Updated to take a non pointer.